README.md

```diff
@@ -11,6 +11,18 @@ This tool is currently in beta, while any issues are worked through. Please feel
 
 This is intended to be used as a stop-gap for teams who are using Kotlin in a legacy system that started out in Java. New projects or projects that have the ability to migrate to newer tools made for Kotlin from the ground up should do so. [Kotlin Serialization](https://github.com/Kotlin/kotlinx.serialization) and [Moshi](https://github.com/square/moshi) are great examples of these tools.
 
+## Disclaimer
+This library uses the Kotlin Reflection library and can incur significant performance losses both when constructing the adapter and when converting the JSON to a model. This incurs a large penalty on performance and can adversely affect your application. It is highly suggested that performance tests be done before using this library to ensure that the performance losses are acceptable for your specific application.
```

Comment:
> This incurs a large penalty on performance and can adversely affect your application. 

I don't think that sentence is needed at all...seems to be redundant with the previous one. You should also be very explicit about the fact that the performance cost is relative to the _Java reflection_ that that default Gson adapter uses.

>  It is highly suggested that performance tests be done before using this library to ensure that the performance losses are acceptable for your specific application.

"highly" should probably be "strongly". In fact for that whole sentence maybe we should try something more like

> It is strongly suggested that profiling be done in your own application when first including this library.